crypto/attachment.go

```diff
@@ -13,6 +15,178 @@ import (
 	"github.com/pkg/errors"
 )
 
+// AttachmentProcessor2 keeps track of the progress of encrypting an attachment
+// (optimized for encrypting large files).
+// With this processor, the caller has to first allocate
+// a buffer large enough to hold the whole data packet.
+type AttachmentProcessor2 struct {
+	keyPacket        []byte
+	dataLength       int
+	plaintextWriter  io.WriteCloser
+	ciphertextWriter *io.PipeWriter
+	err              error
+	done             sync.WaitGroup
+}
+
+// GetKeyPacket returns the key packet for the attachment.
+// This should be called only after Finish() has been called.
+func (ap *AttachmentProcessor2) GetKeyPacket() []byte {
+	return ap.keyPacket
+}
+
+// GetDataLength returns the number of bytes in the DataPacket.
+// This should be called only after Finish() has been called.
+func (ap *AttachmentProcessor2) GetDataLength() int {
+	return ap.dataLength
+}
+
+// Process writes attachment data to be encrypted.
+func (ap *AttachmentProcessor2) Process(plainData []byte) error {
+	defer runtime.GC()
+	_, err := ap.plaintextWriter.Write(plainData)
+	return errors.Wrap(err, "gopenpgp: couldn't write attachment data")
+}
+
+// Finish tells the processor to finalize encryption.
+func (ap *AttachmentProcessor2) Finish() error {
+	defer runtime.GC()
+	if ap.err != nil {
+		return ap.err
+	}
+	if err := ap.plaintextWriter.Close(); err != nil {
+		return errors.Wrap(err, "gopengpp: unable to close plaintext writer")
+	}
+	if err := ap.ciphertextWriter.Close(); err != nil {
+		return errors.Wrap(err, "gopengpp: unable to close the dataPacket writer")
+	}
+	ap.done.Wait()
+	if ap.err != nil {
+		fmt.Printf("Error while finishing %v\n", ap.err)
```

Comment:
Is this intended or a leftover from debugging?

Reply:
It's from debugging yes.